sdk/tables/azure-data-tables/samples/async_samples/sample_conditional_update_async.py

```diff
@@ -0,0 +1,115 @@
+            "product_id": uuid4()
+        }
+        self.entity2 = {
+            "PartitionKey": "color",
+            "RowKey": "brand",
+            "color": "Red",
+            "inventory_count": 42,
+            "barcode": b"135aefg8oj0ld58" # cspell:disable-line
+        }
+
+    async def conditional_update_basic(self):
+        async with TableClient.from_connection_string(self.connection_string, self.table_name_prefix + uuid4().hex) as table_client:
+            await table_client.create_table()
+            metadata1 = await table_client.create_entity(entity=self.entity1)
+            print("Entity:")
+            print(self.entity1)
+
+            # Merge properties of an entity with one that already existed in a table
+            try:
+                await table_client.update_entity(entity=self.entity2, mode=UpdateMode.MERGE, match_condition=MatchConditions.IfNotModified, etag=metadata1["etag"])
+            except ResourceModifiedError:
+                print("This entity has been altered and may no longer be in the expected state.")
+            entity2 = await table_client.get_entity(partition_key=self.entity1["PartitionKey"], row_key=self.entity1["RowKey"])
+            print("Entity after merge:")
+            print(entity2)
+
+            # Replace properties of an entity with one that already existed in a table
+            try:
+                table_client.update_entity(entity=self.entity2, mode=UpdateMode.REPLACE, match_condition=MatchConditions.IfNotModified, etag=entity2.metadata["etag"])
+            except ResourceModifiedError:
+                print("This entity has been altered and may no longer be in the expected state.")
+            entity3 = await table_client.get_entity(partition_key=self.entity1["PartitionKey"], row_key=self.entity1["RowKey"])
+            print("Entity after replace:")
+            print(entity3)
+
+            await table_client.delete_table()
+
+    async def conditional_update_with_a_target_filed(self):
+        async with TableClient.from_connection_string(self.connection_string, self.table_name_prefix + uuid4().hex) as table_client:
+            await table_client.create_table()
+            await table_client.create_entity(entity=self.entity1)
+            target_field = "barcode"
+
+            # In this scenario, will try to create an entity at first. If the entity with the same PartitionKey and RowKey already exists,
+            # will update the existing entity when target field is missing.
+            try:
+                await table_client.create_entity(entity=self.entity2)
+            except ResourceExistsError:
+                entity = await table_client.get_entity(partition_key=self.entity2["PartitionKey"], row_key=self.entity2["RowKey"])
+                if target_field not in entity:
+                    await table_client.update_entity(entity=self.entity2, mode=UpdateMode.MERGE, match_condition=MatchConditions.IfNotModified, etag=entity.metadata["etag"])
```

Comment:
I wouldn't use `entity=self.entity2` here, because our intention is not to update the whole entity, rather, we just want to add the missing field. So I would pass in _only_ the target field.
```python
await table_cient.update_entity(
    entity={"PartitionKey": self.entity2["PartitionKey"], "RowKey": self.entity2["RowKey"], target_field: "foo"},
    mode=UpdateMode.MERGE,
    match_condition=MatchConditions.IfNotModified,
    etag=entity.metadata["etag"]
)
```